tuist/tuist
# Adding support for resource folder references

Resolves https://github.com/tuist/tuist/issues/202

### Short description 📝

Resources to include in an application can be in the form of files as well as folder references.

<img width="347" alt="resource-references" src="https://user-images.githubusercontent.com/11914919/55674630-f76a3f80-58ae-11e9-8e8c-433c81f07966.png">


### Solution 📦

Add support for folder references within target resources. For consistency, we can use `FileElement` which has already been used in `Project` and `Workspace` to define additional files.

e.g.

```swift
import ProjectDescription

let project = Project(
    name: "MainApp",
    targets: [
        Target(
            name: "App",
            platform: .iOS,
            product: .app,
            bundleId: "io.tuist.App",
            infoPlist: "Config/App-Info.plist",
            sources: "Sources/**",
            resources: [
                "Resources/*.png",
                .folderReference(path: "Examples")
            ])
      ]
)
```

### Implementation 👩‍💻👨‍💻

- [x] Include fixture
- [x] Update manifest
- [x] Update models & Generator

### Test Plan ⚒

- Verify unit tests pass via `swift test`
- Verify acceptance tests pass via `bundle exec rake features`

- Run `tuist generate` within `fixtures/ios_app_with_framework_and_resources`
- Inspect the generated workspace
- Verify `MainApp` has a folder reference to `Examples` which is also included in the **Copy Bundle Resources** phase
